### PR TITLE
Refs [TASK][28400] Write fixer recycler view

### PR DESCRIPTION
**Pull request by aug16vcc**
## Related Tickets

- [#Task_28400](https://edu-redmine.sun-asterisk.vn/issues/28400)

## WHAT
-Write fixer recycler view
## HOW
-Write fixer recycler view
## Review Checklist

| Category      | View Point                                                   | Description                                                  | Expected Reviewer Answer | Vu Chi Cong  | Reviewer2 (name)   |
| ------------- | ------------------------------------------------------------ | ------------------------------------------------------------ | ------------------------ | ------------------ | ------------------ |
| General       | Code should be self-explanatory.                             | Get a feel of story reading, while going through the code. Use appropriate name for variables, functions and classes. If you are taking more time to understand the code, then either code needs refactoring or at least comments have to be written to make it clear. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Is this function or class too big? If yes, is the function or class having too many responsibilities? |                                                              | NO                       | <li>- [x] no</li>  | <li>- [ ] no</li>  |
| General       | Does the code follow DRY?: Do not Repeat Yourself            | Is there any redundant or duplicate code? DRY (Do not Repeat Yourself) principle: The same code should not be repeated more than twice. Consider reusable services, functions and components. Consider generic functions and classes. | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Exception handling and cleanup (dispose) resources           |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| General       | Does the code build and execute correctly?                   |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Redmine       | Does the ticket follow Sun* Redmine working process?         | https://github.com/framgia/Training-Guideline/blob/master/WorkingProcess/redmine/redmine.md | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |
| Documentation | Is there any incomplete code? If so, should it be removed or flagged with a suitable marker like ‘TODO’? |                                                              | YES                      | <li>- [x] yes</li> | <li>- [ ] yes</li> |

### CODE REVIEW

**Review by quockhanhng — APPROVED** [review]

File: app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+        ItemSummaryUserBinding.inflate(container.inflater, container, false)
+    ) {
+
+        // FIXME: RecyclerView inside NestedScrollView causes RecyclerVew to inflate all elements
+        // NOTE: https://www.reddit.com/r/androiddev/comments/d8gi9v/recyclerview_inside_nestedscrollview_causes/f1blxes/
+        override fun onBind(item: SummaryUser, binding: ItemSummaryUserBinding): Unit =
+            with(binding.layoutSummaryUser) {
+                user = item
+                executePendingBindings()
+                Log.d("UserVH", "onBind: ${item.id}")
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt]: log gì đây nhỉ

File: app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        holder.bind(getItem(position))
+    }
+
+    class UserVH(container: ViewGroup) : BaseViewHolder<SummaryUser, ItemSummaryUserBinding>(
+        ItemSummaryUserBinding.inflate(container.inflater, container, false)
+    ) {
+
+        // FIXME: RecyclerView inside NestedScrollView causes RecyclerVew to inflate all elements
+        // NOTE: https://www.reddit.com/r/androiddev/comments/d8gi9v/recyclerview_inside_nestedscrollview_causes/f1blxes/
+        override fun onBind(item: SummaryUser, binding: ItemSummaryUserBinding): Unit =
```
Comment (huytq-2422) [inline app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt]: Unit có thể lược bỏ mà bạn 
```suggestion
        override fun onBind(item: SummaryUser, binding: ItemSummaryUserBinding) =
```

Comment (huytq-2422) [reply app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt]: https://kotlinlang.org/docs/reference/coding-conventions.html#avoiding-redundant-constructs

File: app/src/main/java/com/sunasterisk/thooi/util/ViewExtensions.kt [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package com.sunasterisk.thooi.util
+
+import android.view.LayoutInflater
+import android.view.View
+
+/**
+ * Created by Cong Vu Chi on 06/09/20 23:18.
+ */
+val View.inflater: LayoutInflater
+    get() = LayoutInflater.from(context)
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/thooi/util/ViewExtensions.kt:10]:
```suggestion
fun ViewGroup.inflate(@LayoutRes layoutResource: Int) = LayoutInflater.from(context).inflate(layoutResource, this, false)
```

File: app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import com.sunasterisk.thooi.databinding.ItemSummaryUserBinding
+import com.sunasterisk.thooi.ui.post.detail.SummaryUserAdapter.UserVH
+import com.sunasterisk.thooi.util.inflater
+
+/**
+ * Created by Cong Vu Chi on 06/09/20 23:08.
+ */
+class SummaryUserAdapter : ListAdapter<SummaryUser, UserVH>(SummaryUser.diffUtil) {
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = UserVH(parent)
```
Comment (thonc-2135) [inline app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt:16]: có nên viết base cho tiện k nhỉ

File: app/src/main/java/com/sunasterisk/thooi/util/ViewExtensions.kt (hunk 3 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/thooi/util/ViewExtensions.kt:10]: Nope. T chỉ cần `LayoutInflater` thôi

File: app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt (hunk 4 above)
Comment (aug16vcc) [reply app/src/main/java/com/sunasterisk/thooi/ui/post/detail/SummaryUserAdapter.kt:16]: Pull sau ô nhé

**Pull request merged**